mule-extensions-api/src/main/java/org/mule/runtime/extension/api/model/AbstractComponentModel.java

```diff
@@ -27,93 +26,47 @@
 public abstract class AbstractComponentModel extends AbstractParameterizedModel
     implements ComponentModel {
 
-  private final OutputModel output;
-  private final OutputModel outputAttributes;
-  private final boolean transactional;
-  private final boolean requiresConnection;
-  private final boolean supportsStreaming;
-  private final Set<Stereotype> stereotypes;
+  private final Set<StereotypeModel> stereotypes;
+  private final List<? extends NestableElementModel> nestedComponents;
 
   /**
    * Creates a new instance
    *
    * @param name                 the model's name
    * @param description          the model's description
    * @param parameterGroupModels a {@link List} with the source's {@link ParameterGroupModel parameter group models}
-   * @param output               an {@link OutputModel} which represents the component's output content
-   * @param outputAttributes     an {@link OutputModel} which represents the component's attributes on the output {@link Message}
-   * @param requiresConnection   whether this component requires connectivity
-   * @param transactional        whether this component supports transactions
-   * @param supportsStreaming    whether this component supports streaming
    * @param displayModel         a model which contains directive about how this component is displayed in the UI
-   * @param stereotypes          A {@link Set} of {@link Stereotype stereotypes}
+   * @param stereotypes          A {@link Set} of {@link StereotypeModel stereotypes}
    * @param modelProperties      A {@link Set} of custom properties which extend this model
+   * @param nestedComponents     a {@link List} with the components contained by this model
    * @throws IllegalArgumentException if {@code name} is blank
    */
   protected AbstractComponentModel(String name,
                                    String description,
                                    List<ParameterGroupModel> parameterGroupModels,
-                                   OutputModel output,
-                                   OutputModel outputAttributes,
-                                   boolean requiresConnection,
-                                   boolean transactional,
-                                   boolean supportsStreaming,
                                    DisplayModel displayModel,
-                                   Set<Stereotype> stereotypes,
-                                   Set<ModelProperty> modelProperties) {
+                                   Set<StereotypeModel> stereotypes,
+                                   Set<ModelProperty> modelProperties,
+                                   List<? extends NestableElementModel> nestedComponents) {
```

Comment:
convention is that model properties are always the last thing. I would actually put this below parameterGroups